spec.html

```diff
@@ -23920,7 +23920,9 @@ <h1>Runtime Semantics: ClassDefinitionEvaluation</h1>
               1. Return ? Construct(_func_, _args_, NewTarget).
             1. Else,
               1. NOTE: This branch behaves similarly to `constructor() {}`.
-              1. Return ? OrdinaryCreateFromConstructor(NewTarget, *"%Object.prototype%"*).
+              1. Let _result_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%Object.prototype%"*).
+              1. Perform ? InitializeInstanceElements(_result_, _F_).
```

Comment:
This needs to be done in the other branch too.

And at that point we might as well combine them, as in

```
  4. Let _result_ be ? Construct(_func_, _args_, NewTarget).
v. Else,
  1. NOTE: This branch behaves similarly to `constructor() {}`.
  2. Let _result_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%Object.prototype%"*).
vi. Perform ? InitializeInstanceElements(_result_, _F_).
vii. Return _result_.

Reply:
in the other branch, i believe `InitializeInstanceElements` should not be called as it is called from `SuperCall` runtime semantics.

Reply:
I don't understand what that means. Where would we be invoking the SuperCall runtime semantics?

The idea of the other branch is that it's supposed to be doing basically the same things as `constructor(...args) { super(...args); }`. We have half of [what `super(...args)` does](https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#sec-super-keyword-runtime-semantics-evaluation), namely step 6 (of the third of the three algorithms linked). But we don't have the other half, which is step 8, the InitializeInstanceElements call.

I don't know how else InitializeInstanceElements would get called in that branch.

Reply:
you're right i'm dumb and i had it backwards. will update.